doc/source/user/workbench.rst

```diff
@@ -0,0 +1,178 @@
+(where all templates and extra files are located).
+
+Plugins
+~~~~~~~
+
+The `plugins` block is required. Within the `plugins` blocks are
+`plugin` sub-blocks where WATTS plugins are defined.
+
+.. code-block:: text
+    plugins{
+        plugin(ID1){
+            code = moose
+            template = "sam_template"
+            exec_dir = SAM_DIR
+            exec_name = "sam-opt"
+            extra_inputs=["file1", "file2", "file3"]
+            extra_templat_inputs=["template1", "template2", "template3"]
+            show_stderr = False
+            show_stdout = False
+        }
+
+        plugin(ID2){
+            code = PyARC
+            template = "pyarc_template"
+            executable = "path/to/pyarc/executable"
+            show_stderr = False
+            show_stdout = False
+        }
+    }
+
+Multiple `plugin` sub-blocks can be defined within the `plugins` block where
+each plugin is given a unique identity, as represented by `(ID1)` and `(ID2)`
+in the snippet above. For each sub-block the basic inputs are::
+
+     `code` : Name of the application
+     `template` : Name of the template file
+     `show_stderr` : Option to display error
+     `show_stdout` : Option to display output
+     `exec_dir` : Environment variable that points to the directory in which the application's executable is stored
+     `extra_inputs` : Additional non-templated input files
+     `extra_templateinputs` : Additional tempalted input files
+     `exec_name` : Name of the executable
+     `executable` : Path to the executable
+
+Note that user must provide either BOTH `exec_dir` and `exec_name` or
+ONLY `executable`. Additionally, there are  application-specific inputs
+that can be provided to the plugins such as::
+
+    `extra_args` (multiple apps) : Extra arguments to applications
+    `plotfl_to_csv` (RELAP5) : Option to convert `plotfl` file to `csv` file
+    `conv_channel` (SAS) : Path to `CHANNELtoCSV.x`
+    `conv_primar4` (SAS) : Path to `PRIMAR4toCSV.x`
+    `auto_link_files` (Dakota) : List of files for Dakota to link automatically
+    `scores` (OpenMC) : List of filters for tallies
+    `score_names` (OpenMC) : List of user-given names for tally filters
+
+Workflow level
+~~~~~~~~~~~~~~
+
+The `workflow_level1` block is required. The plugin to be used is specified
+by the `plugin` keyword::
+
+    plugin = ID1
+
+where 'ID1' is the ID of the plugin provided in the `plugins` block. The
+`variable` sub-block is where the values of the templated variables are
+specified, as shown below::
+
+    variables{
+        param(T_inlet) {value = 873.15}
+        param(T_outlet) {value = 1000.15}
+        param(flow_area) {value = 3.14e-4}
+        param(component_name) {value = pipe1}
+        param(Dh) {value = 1 unit = m}
+        param(location) {list = [top bottom left right]}
+        param(length) {list = [3.0 5.0 7.0 10.0]}
+        param(T_mean) {func = "0.5 * ( T_inlet + T_outlet )"}
+        param(wall_bc) {bool = "True"}
```

Comment:
We should develop different param templates that show the different options (param.value.tmpl, param.list.tmpl, param.func.tmpl, param.bool.tmpl, etc.)